packages/flutter/test/semantics/semantics_update_test.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('Semantics update', () async {
+    final SemanticsUpdateTestBinding binding = SemanticsUpdateTestBinding();
+    binding.pipelineOwner.ensureSemantics();
+    await binding.runTest(() async {
+      binding.attachRootWidget(
+        MaterialApp(
+          home: MergeSemantics(
+            child: Semantics(
+              value: 'test 1',
+              textField: true,
+              // This semantics b
+              child: MergeSemantics(
+                child: Semantics(
+                  value: 'test 2',
+                  textField: true,
+                  child: const Text('test 3'),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+      SemanticsUpdateBuilderSpy.observations.clear();
+      binding.scheduleFrame();
+      await binding.pump(null, EnginePhase.sendSemanticsUpdate);
+
+      // Checks all nodes are connected.
+      // Starts with root node.
+      int currentNode = 0;
+      while(SemanticsUpdateBuilderSpy.observations.containsKey(currentNode)) {
```

Comment:
Does this while loop actually run? I would have expected that `SemanticsUpdateBuilderSpy.observations` is empty because you cleared it above and then pumped a frame that didn't change any semantics?

Reply:
Can you add an expect before the while loop to check that `SemanticsUpdateBuilderSpy.observations` actually contains the expected number of nodes?

Reply:
Instead of doing the while loop I'd also have more confidence in the test if it would just check individually that all expected nodes are in `observations`.